### ISSUE TITLE
Create directory by google_storage_bucket_object is not working "gsutil rsync" command

### ISSUE THREAD

**Issue by dddwsd** [hashicorp/terraform-provider-google#15605]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

```bash
Terraform v1.5.5
on darwin_arm64
```

### Affected Resource(s)

* google_storage_bucket_object

### Terraform Configuration Files

```tf
# gsutil rsync failed
resource "google_storage_bucket_object" "train-template_base_terraform_object" {
  name   = "train-template_base_terraform/" # folder name should end with '/'
  content = " "            # content is ignored but should be non-empty
  bucket = "dddwsd-test"
}

# gsutil rsync succeed
resource "google_storage_bucket_object" "train-template_base_terraform2_object" {
  name   = "train-template_base_terraform2/empty"
  source = "empty"
  bucket = "dddwsd-test
}
```

### Panic Output

 ~/test  gsutil rsync -r gs://dddwsd-test/train-template_base_terraform terraform/

WARNING: gsutil rsync uses hashes when modification time is not available at
both the source and destination. Your crcmod installation isn't using the
module's C extension, so checksumming will run very slowly. If this is your
first rsync since updating gsutil, this rsync can take significantly longer than
usual. For help installing the extension, please see "gsutil help crcmod".

Building synchronization state...
Starting synchronization...
Copying gs://lgair-mlops-train/train-template_base_terraform/...
Skipping attempt to download to filename ending with slash
(terraform/). This typically happens when using gsutil to download
from a subdirectory created by the Cloud Console
(https://cloud.google.com/console)
InvalidUrlError: Invalid destination path: terraform/.

### Expected Behavior

in both gcs object, "gsutil rsync" operation should have completed successfully.

### Actual Behavior

only "train-template_base_terrafom2" object has successfully finished "gsutil rsync"

### Steps to Reproduce

1. `terraform apply`
2. `mkdir terraform`
3. `gsutil rsync -r gs://${bucket_name}/train-template_base_terraform terraform/`

### Important Factoids

1. terraform must be connected with gcp.
2. cloud storage architecture is not same as linux filesystem - https://cloud.google.com/storage/docs/folders?hl=ko#overview
4. If you create a folder in the way it is shown in the official document, "gsutil rsync" does not work. - https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/storage_bucket_object.html#example-usage
(According to google support, the content-length of the created folder is 1(content is " "), so the "gsutil rsync" command recognizes it as a file, not a dir.)
5. if you create a folder with empty file, "gsutil rsync" is working successfully. 
```tf
resource "google_storage_bucket_object" "train-template_base_terraform3_object" {
  name   = "train-template_base_terraform3/empty"
  content = " "
  bucket = google_storage_bucket.lgair_mlops_train.name
}
```

To sum up, I think it would be better to set the content-length to 0 or insert an empty file when creating an empty folder through google_storage_bucket_object.

### References

* #0000

**Comment by edwardmedia** [comment]
b/297175715

**Comment by dddwsd** [comment]
what's the meaning of "b/297175715"?

**Comment by gurusai-voleti** [comment]
@dddwsd the terraform resource is working as expected, it successfully created an empty folder for the specified terraform config, the sync might be an issue with the gsutil rysnc command, please file a bug for the gsutil tool

@trodge  (oncall) this user reported is not an issue and working as expected, can you help close the ticket as I don't have permission to close the ticket
